### PR TITLE
Consistent treatment of energy deposition for photon-electron calculations

### PR DESCRIPTION
**Pull request by paulromano**
This PR addresses #1453. With the changes here, we can now accurately tally heating from neutrons, photons, electrons, and positrons. As proposed in #1453, I've eliminated the precomputed photon heating numbers since they are not accurate when thick-target bremsstrahlung is enabled (the default). Now, when you have a tally that asks for `heating`, the estimator is automatically set to a collision estimator, although under the hood it's really still an analog estimator (the collision flux estimate is not used). I've done it this way (rather than using an analog estimator) so that if you have a tally that has heating and other scores, the collision estimator is used for the other scores. This required that the calls to `score_general_ce` separate the flux from the filter weight -- and honestly, I think it's cleaner and more consistent this way.

If you look at the logic for the (non-neutron) heating score now, you'll see it's very simple. There's no more messing around to figure out what the particle is or having a special case for pair production. It _just works_.

The simple problem outlined in #1453 (monoenergetic photon source of energy E in an infinite medium) now gets the result you'd expect (regardless of whether TTB is on): the energy deposition per source is exactly the source energy.

One final note- the cases where test results changed are due to solely to the change in how heating is calculated.

### CODE REVIEW

File: src/output.cpp [hunk 1]
```diff
@@ -527,8 +527,8 @@ std::pair<double, double>
 mean_stdev(const double* x, int n)
 {
   double mean = x[static_cast<int>(TallyResult::SUM)] / n;
-  double stdev = n > 1 ? std::sqrt((x[static_cast<int>(TallyResult::SUM_SQ)]/n
-    - mean*mean)/(n - 1)) : 0.0;
+  double stdev = n > 1 ? std::sqrt(std::max(0.0, (
+    x[static_cast<int>(TallyResult::SUM_SQ)]/n - mean*mean)/(n - 1))) : 0.0;
```
Comment (amandalund) [inline src/output.cpp]: Can the variance be negative?

File: openmc/data/photon.py [hunk 2]
```diff
@@ -929,83 +926,6 @@ def _add_bremsstrahlung(self):
...
-
-        # Photoelectric effect
-        if 522 in self:
-            # Account for fluorescent photons
-            for mt, rx in self.reactions.items():
-                if mt >= 534 and mt <= 572:
-                    shell = _REACTION_NAME[mt][1]
-                    relax_data = self.atomic_relaxation
-                    if relax_data is not None:
-                        e_f = relax_data.energy_fluorescence(shell)
```
Comment (amandalund) [inline openmc/data/photon.py]: Is there still a situation where we might want to calculate the expected energy of the fluorescence photons now that we’re not storing the heating numbers (i.e., is there a reason to keep the method `energy_fluorescence`)?

File: src/output.cpp (hunk 1 above)
Comment (paulromano) [reply src/output.cpp]: I ran into this when using the analog heating estimator. In the case of a monoenergetic photon source, every particle scores exactly the energy it started with, so the variance is expected to be zero. However, due to limited precision and roundoff errors, sometimes 0 ends up being some very small negative number, which causes problems here.

Comment (amandalund) [reply src/output.cpp]: gotcha, makes sense!

File: openmc/data/photon.py (hunk 2 above)
Comment (paulromano) [reply openmc/data/photon.py]: Good point. That method is no longer used anywhere, so I'll go ahead and remove it. If we ever want/need to revive it, it's still there in the git history :)

Comment (paulromano) [reply openmc/data/photon.py]: :heavy_check_mark:

**Pull request merged**